src/vmm/src/vstate/vm.rs

```diff
@@ -196,7 +201,7 @@ impl Vm {
         kvm_max_memslots: usize,
         track_dirty_pages: bool,
     ) -> Result<()> {
-        if guest_mem.num_regions() > kvm_max_memslots {
+        if guest_mem.num_regions() + self.next_memory_slot as usize > kvm_max_memslots {
```

Comment:
I think `next_memory_slot` should be 0 here, and otherwise an error should be returned. This function is called before any devices are added so nobody should have incremented it.